packages/terra-form/src/Select.jsx

```diff
@@ -62,7 +74,7 @@ const Select = ({
       defaultValue={defaultValue}
       className={classNames('terra-Form-select', additionalSelectFieldProps.className)}
     >
-      {choices.map(choice => <option key={choice.toString()} value={choice}>{choice}</option>)}
```

Comment:
Updated to be a combination of value and display parameters, converted to a string.